local-setup-testing/test/main.test.ts

```diff
@@ -2,39 +2,72 @@ import { expect } from 'chai';
 import { Wallet, Contract, utils } from 'zksync-web3';
 import * as hre from 'hardhat';
 import { ethers } from 'ethers';
-import { Deployer } from "@matterlabs/hardhat-zksync-deploy";
+import { Deployer } from '@matterlabs/hardhat-zksync-deploy';
 
-const RICH_WALLET_PK = '0x7726827caac94a7f9e1b160f7ea819f172f7b6f9d2a97f992c38edeab82d4110';
+const RICH_WALLET_PK =
+  '0x7726827caac94a7f9e1b160f7ea819f172f7b6f9d2a97f992c38edeab82d4110';
 
-async function deployGreeter(deployer: Deployer): Promise<Contract> {
-    const artifact = await deployer.loadArtifact("Greeter");
-    return await deployer.deploy(artifact, ["Hi"]);
-}
+let greeter: Contract;
+
+async function deployGreeter(deployer: Deployer) {
+  try {
+    console.log('Deploying contract');
+    const artifact = await deployer.loadArtifact('Greeter');
 
-describe("Greeter", function () {
-    let deployer: Deployer;
+    // return await deployer.deploy(artifact, ['Hi']);
+    greeter = await deployer.deploy(artifact, ['Hi']);
+  } catch (error) {
+    console.error('Error deploying contract');
+    console.error(error);
+    throw new Error('Error deploying contract');
+  }
+}
 
-    before("Fund the wallet", async () => {
-        deployer = new Deployer(hre, new Wallet(RICH_WALLET_PK));
+describe('Greeter', function () {
+  let deployer: Deployer;
 
-        const depositHandle = await deployer.zkWallet.deposit({
-            to: deployer.zkWallet.address,
-            token: utils.ETH_ADDRESS,
-            amount: ethers.utils.parseEther('0.001'),
-        });
+  before('Fund the wallet', async () => {
+    deployer = new Deployer(hre, new Wallet(RICH_WALLET_PK));
 
-        await depositHandle.wait();
+    const depositHandle = await deployer.zkWallet.deposit({
+      to: deployer.zkWallet.address,
+      token: utils.ETH_ADDRESS,
+      amount: ethers.utils.parseEther('0.001'),
     });
 
-    it("Should return the new greeting once it's changed", async () => {
-        const greeter = await deployGreeter(deployer);
+    await depositHandle.wait();
+    await deployGreeter(deployer);
+  });
 
-        expect(await greeter.greet()).to.eq("Hi");;
+  it('Deployer should be valid address', async () => {
+    expect(deployer.zkWallet.address).to.be.properAddress;
+  });
 
-        const setGreetingTx = await greeter.setGreeting("Hola, mundo!");
-        await setGreetingTx.wait();
+  it("Should return the new greeting once it's changed", async () => {
+    // const greeter = await deployGreeter(deployer);
```

Comment:
Commented code

Reply:
fixed in 070e48285b7b7f49847a2bb180adb57ca86b7468